xarray_beam/_src/core.py

```diff
@@ -307,13 +324,36 @@ def __init__(
     self.sharded_dim = max(lengths, key=lengths.get) if lengths else None
     self.shard_count = self._shard_count()
 
+  @property
+  def _first(self) -> xarray.Dataset:
+    return self._datasets[0]
+
+  @property
+  def _datasets(self) -> List[xarray.Dataset]:
+    if type(self.dataset) is xarray.Dataset:
+      return [self.dataset]
+    return list(self.dataset)
+
+  def _validate(self, dataset, split_vars):
+    """Raise errors if input parameters are invalid."""
+    if not dataset and not type(xarray.Dataset):
```

Comment:
`type(xarray.Dataset)` looks wrong. Maybe you want something like this?
```suggestion
    if not isinstance(dataset, xarray.Dataset) and not dataset:
```